wordpress/wp-content/themes/ws-custom-theme/assets/js/script.js

```diff
@@ -14,47 +14,58 @@ CUSTOM.core = function () {
             $('#table-id').append( '</table>' );
             self.clickFunction();
             var ajaxUrl = 'http://localhost/wordpress/wp-json/mypost/api/get-posts/';
+            var termdetails=[];
             jQuery.ajax({
                 url: ajaxUrl,
                 type: 'GET',
                 success: function (response) {
                     var jsonResponse = JSON.parse(response);
-                    var categorynamelist=new Array();
                     $.each(jsonResponse,function(postArray,post){
-                        var postcategory = post.post_category;
                         var postid = post.post_id;
                         var postname = post.post_name;
                         var postdescription = post.post_description;
                         var postdate = post.post_date;
                         var poststatus = post.post_status;  
                         var posttermid = post.post_term_id;
-                        console.log(posttermid);
-                        categorynamelist.push(postcategory);                        
-                        $('#data').append("<tr class='data-row'><td>"+postid+"</td><td>"+postdate+"</td><td>"+postname+"</td><td>"+postcategory+"</td><td>"+postdescription+"</td><td>"+poststatus+"</td></tr>");
-                    });
-                    var categoryname = categorynamelist.filter(function(element, index, self) {
-                        return index === self.indexOf(element);
-                    });                   
-                    $.each(categoryname,function(categories,categorys){
-                        var li = $('<li><input type="checkbox" name="' + categories + '" id="' + categorys + '"term-d"'+ '"/>' + '<label for="' + categorys + '"></label></li>');
-                        li.find('label').text(categorys);
-                        $('.div-buttons').append(li);
-                        self.categoryFunction(categorys);                     
+                        var postcategory = post.post_category;
+                        if(termdetails[post.post_term_id]===undefined){
+                            var termid=post.post_term_id;
+                            var termname=post.post_category
+                            termdetails[termid]=termname;
```

Comment:
termdetails[post.post_term_id]=post.post_category;
 
 try to use minimum variables 
 don't declare variable until it is used more than one time